src/atoms/typography/heading-icon.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { HeadingPriority } from "../constants/heading-priority";
+import { Heading } from "./heading";
+import { Icons } from "../constants/icons";
+import { IconSizes } from "../constants/icon-sizes";
+import { Icon } from "../icons/icon";
+
+// -------------------------------------------------------------------------------------------------
+// #region Interfaces
+// -------------------------------------------------------------------------------------------------
+
+export interface HeadingIconProps {
+    children?: any;
```

Comment:
Nitpick - prefer wrapping this interface with `PropsWithChildren` instead of manually defining `children`